lib/base-compiler.js

```diff
@@ -73,13 +76,17 @@ Compile.prototype.exec = function (compiler, args, options) {
     return exec.execute(compiler, args, options);
 };
 
-Compile.prototype.runCompiler = function (compiler, options, inputFilename) {
-    return this.exec(compiler, options, {
-        timeoutMs: this.env.gccProps("compileTimeoutMs", 100),
-        maxErrorOutput: this.env.gccProps("max-error-output", 5000),
-        env: this.env.getEnv(this.compiler.needsMulti),
-        wrapper: this.env.compilerProps("compiler-wrapper")
-    }).then(function (result) {
+Compile.prototype.runCompiler = function (compiler, options, inputFilename, execOptions) {
+    if (typeof(execOptions)==='undefined') {
```

Comment:
Actually a better approach to this might be to have a get defaultOptions() then use http://underscorejs.org/#extend to add your max options.